### PR TITLE
Added the augment broom dialog use models

### LINKED ISSUE #5647: Add broom to the Model > Use Model menu
Initially this is easy, but there will then - or in parallel - be more work to do!  This needs advice from @dannyparsons and David.
broom is designed to use models in a tidy way, so I suggest it becomes the first in the tab order of the packages.
a) There are just 3 commands, namely augment, glance, tidy.  They all work trivially, i.e. initially we just put them in as they stand.  Then, for example after an lm(yield~fert) in the Model > Model dialogue, each can just be used as it stand, 
broom::augment(last_model)
broom::glance(last_model)
broom::tidy(last_model)

The results go into the output window.  That's a good start.
But each produces a tibble (data frame) object.  So the next step will be to be able to save the data frame - as an object - which we can do - and then be able to see it in the grid as a tabble (data frame!)
The augment could hopefully be a separate data frame, or add variables to the data frame that produced the model - that's what broom is designed to do.

At the same time there are some minor edits to the Use Model dialogue:
a) It is too wide and that looks messy.  Let's make narrower.
b) It is too short, so the last row is cut off.  Make it a bit longer.
c) More difficult - the Help on General doesn't work - no package called General!  This is a bit more complicated, because there are multiple packages used here.  I am not sure what to do here - it needs a bit of discussion, partly because it should be done quickly!

### PR DESCRIPTION
**Pull request by Wycklife**
Fixes #5647
This is working for a single model.
@africanmathsinitiative/developers This is ready for review

### CODE REVIEW

**Comment by rdstern** [comment]
This dialogue starts to implement the broom augment function.  
@volloholic may be able to confirm, but I think this receiver should change to be a single receiver.  It is producing mode variables at the original data frame level.  

Currently the output is to the results window only.  It should now also be able to write back to the original data frame.

This is unlike the other 2 dialogues that will write to a new data frame.  This dialogue will add variables to the existing data frame.  There is also a question of whether the dialogue is in the right place - perhaps it should be in the Fit model section.  Perhaps it should even be "behind the scenes" and become part of the saving results from the fit-model dialogues.

So, leave this for now, until we resolve these ideas.

**Comment by Wycklife** [comment]
@rdstern Could you please check if this is okay.

**Comment by Wycklife** [comment]
@rdstern .Hi Roger, @dannyparsons suggested that I should leave the two check boxes in the Augment dialog for now, since the the dialog is only adding extra columns to the existing data frame.

File: instat/dlgAugment.vb [hunk 1]
```diff
@@ -1,3 +1,91 @@
...
+        SetRCodeForControls(True)
+        TestOKEnabled()
+    End Sub
+
+    Private Sub CoreControls_ControlContentsChanged() Handles ucrModelReceiver.ControlContentsChanged ', ucrSaveNewDataFrame.ControlContentsChanged
+        TestOKEnabled()
+    End Sub
+
+    Private Sub UpdateAssignTo()
+        ucrBase.clsRsyntax.SetAssignTo(strAssignToName:="augment_data", strTempDataframe:=ucrModelSelector.ucrAvailableDataFrames.cboAvailableDataFrames.Text, bAssignToColumnWithoutNames:=True)
```
Comment (shadrackkibet) [inline instat/dlgAugment.vb]: I think this can be done on SetDefaults()

File: instat/dlgAugment.vb [hunk 2]
```diff
@@ -1,3 +1,91 @@
...
+        clsAugment.SetRCommand("augment")
+        clsAugment.SetPackageName("broom")
+        ucrBase.clsRsyntax.SetBaseRFunction(clsAugment)
+        UpdateAssignTo()
+    End Sub
+
+    Public Sub SetRCodeForControls(bReset As Boolean)
+        'ucrSaveNewDataFrame.SetRCode(ucrBase.clsRsyntax.clsBaseCommandString, bReset)
+        ucrModelReceiver.SetRCode(clsAugment, bReset)
+        UpdateAssignTo()
```
Comment (shadrackkibet) [inline instat/dlgAugment.vb]: Since you have one RFunction then this could be simplified.

File: instat/dlgAugment.vb [hunk 3]
```diff
@@ -1,3 +1,91 @@
-﻿Public Class dlgAugment
+﻿
+Imports instat.Translations
+Public Class dlgAugment
+    Public bfirstload As Boolean = True
+    Public bReset As Boolean = True
+    Private bResetSubdialog As Boolean = False
+    Private clsRSyntax As RSyntax
```
Comment (shadrackkibet) [inline instat/dlgAugment.vb]: I don't think this is needed.

File: instat/dlgAugment.vb [hunk 4]
```diff
@@ -1,3 +1,91 @@
...
+
+        ucrModelReceiver.SetParameter(New RParameter("x", 0))
+        ucrModelReceiver.Selector = ucrModelSelector
+        ucrModelReceiver.SetMeAsReceiver()
+
+
+    End Sub
+
+    Private Sub SetDefaults()
+
```
Comment (shadrackkibet) [inline instat/dlgAugment.vb]: You should declare New instance of your RFunction here.

**Comment by Wycklife** [comment]
@rdstern This is fixed too

**Comment by rdstern** [comment]
Just the trivial changes for consistency with the other dialogues, namely Add to Data Frame and New Data Frame Name.  Then please merge

**Comment by Wycklife** [comment]
@rdstern Done!

**Pull request merged**